thewalrus/symplectic.py

```diff
@@ -85,16 +85,18 @@ def expand(S, modes, N):
         array: the resulting :math:`2N\times 2N` Symplectic matrix
     """
     M = S.shape[0] // 2
-    S2 = np.identity(2 * N, dtype=S.dtype)
+    S2 = (
+        np.identity(2 * N, dtype=S.dtype)
+        if not issparse(S)
+        else sparse_identity(2 * N, dtype=S.dtype, format="csr")
+    )
 
-    if issparse(S):
+    if issparse(S) and isinstance(S, (coo_array, dia_array, bsr_array)):
```

Comment:
There are some scipy sparse types that you are missing in this comparison -- is that intentional?

Reply:
Yes! Other types (the avoided ones) can use numpy-style indexing.